apps/fluent-tester/src/FluentTester/TestComponents/Avatar/BasicAvatar.macos.tsx

```diff
@@ -0,0 +1,145 @@
+import { SvgIconProps } from '@fluentui-react-native/icon';
+import { MenuPicker } from '../Common/MenuPicker';
+
+type WithUndefined<T> = T | typeof undefinedText;
+
+const avatarActive: AvatarActive[] = ['unset', 'active', 'inactive'];
+const avatarColors: AvatarColor[] = ['neutral', 'brand', 'colorful', ...AvatarColors];
+const avatarActiveAppearance: AvatarActiveAppearance[] = ['ring'];
+
+const allSizes: WithUndefined<AvatarSize>[] = [undefinedText, ...AvatarSizes];
+const allPresences: WithUndefined<PresenceBadgeStatus>[] = [undefinedText, ...PresenceBadgeStatuses];
+
+export const StandardUsage: FunctionComponent = () => {
+  const [isSquare, setSquare] = useState(false);
+  const [showImage, setShowImage] = useState(true);
+  const [outOfOffice, setOutOfOffice] = useState(false);
+  const [active, setActive] = useState<AvatarActive>('unset');
+  const [activeAppearance, setActiveAppearance] = useState<AvatarActiveAppearance>('ring');
+  const [imageSize, setImageSize] = useState<WithUndefined<AvatarSize>>(72);
+  const [presence, setPresence] = useState<WithUndefined<PresenceBadgeStatus>>('available');
+  const [avatarColor, setAvatarColor] = useState<AvatarColor>('brass');
+
+  const onActiveChange = useCallback((value) => setActive(value), []);
+  const onActiveAppearanceChange = useCallback((value) => setActiveAppearance(value), []);
+  const onAvatarColorChange = useCallback((value) => setAvatarColor(value), []);
+  const onSizeChange = useCallback((value) => setImageSize(value), []);
+
+  const onPresenceChange = useCallback((value) => setPresence(value), []);
+
+  const theme = useTheme();
+  const textStyles = { color: theme.colors.inputText as ColorValue };
+  const avatarSizesForPicker = allSizes.map((size) => size.toString());
+
+  const fontBuiltInProps = {
+    fontFamily: 'Arial',
+    codepoint: 0x2663,
+  };
+
+  const svgProps: SvgIconProps = {
+    src: TestSvg,
+    viewBox: '0 0 500 500',
+  };
+
+  const svgIconsEnabled = ['ios', 'macos', 'win32', 'android'].includes(Platform.OS as string);
+
+  return (
+    <View style={commonStyles.root}>
+      <View style={commonStyles.settings}>
+        <View style={commonStyles.switch}>
+          <Text style={textStyles}>Show images</Text>
+          <Switch value={showImage} onValueChange={setShowImage} />
+        </View>
+        <View style={commonStyles.switch}>
+          <Text style={textStyles}>Set square Avatar</Text>
+          <Switch value={isSquare} onValueChange={() => setSquare(!isSquare)} />
+        </View>
+        <View style={commonStyles.switch}>
+          <Text style={textStyles}>Set outOfOffice</Text>
+          <Switch value={outOfOffice} onValueChange={() => setOutOfOffice(!outOfOffice)} />
+        </View>
+
+        <MenuPicker prompt="Size" selected={imageSize.toString()} onChange={onSizeChange} collection={avatarSizesForPicker} />
+        <MenuPicker prompt="Active" selected={active} onChange={onActiveChange} collection={avatarActive} />
```

Comment:
I really don't think we should be duplicating every test like this thrice on each platform. Instead, I think we should have one file for each test (that references <MenuPicker>), and MenuPicker within itself can render a different control based on platform.  

Then, you would be containing the responsibility of knowing what works on what platform into the one control, and client code (aka, all the tests) can use <MenuPicker> without worry and duplication.

Reply:
It seems you already put in the effort to make sure <MenuPicker> has the same API as StyledPicker (BasicAvatar.tsx, line 29). So combining the two should be easy?

Reply:
I see how that can be a big issue. I've gone back and made it so only the control is responsible for choosing what to render.